corehq/apps/app_manager/models.py

```diff
@@ -1275,15 +1276,15 @@ def case_list_modules(self):
     def is_case_list_form(self):
         return bool(self.case_list_modules)
 
-    def get_save_to_case_updates(self, case_type):
+    def get_save_to_case_updates(self):
         """
         Get a flat list of case property names from save to case questions
         """
-        updates = set()
+        updates_by_case_type = defaultdict(set)
         for save_to_case_update in self.case_references_data.get_save_references():
-            if save_to_case_update.case_type == case_type:
-                updates |= set(save_to_case_update.properties)
-        return updates
+            case_type = save_to_case_update.case_type
+            updates_by_case_type[case_type] |= set(save_to_case_update.properties)
```

Comment:
nit: I think it's marginally more efficient to do
```py
updates_by_case_type[case_type].update(save_to_case_update.properties)
```
since that does not create a new set (and throw away the old one).